ecl/hqlcpp/hqlttcpp.cpp

```diff
@@ -3821,6 +3818,34 @@ void HqlCppTranslator::convertLogicalToActivities(WorkflowItem & curWorkflow)
 }
 
 //------------------------------------------------------------------------
+static HqlTransformerInfo setResultToExtractTransformerInfo("SetResultToExtractTransformer");
+SetResultToExtractTransformer::SetResultToExtractTransformer()
+: NewHqlTransformer(setResultToExtractTransformerInfo)
+{
+}
+
+
+IHqlExpression * SetResultToExtractTransformer::createTransformed(IHqlExpression * expr)
+{
+    OwnedHqlExpr transformed = PARENT::createTransformed(expr);
+    updateOrphanedSelectors(transformed, expr);
+
+    IHqlExpression * normalized = NULL;
+    if (transformed->getOperator() == no_setresult)
+    {
+        normalized = convertSetResultToExtract(transformed);
```

Comment:
minor:  Better would be 

    OwnedHqlExpr normalized = convert...

which would remove the need for the Release()